library/core/src/ptr/non_null.rs

```diff
@@ -1821,12 +1857,76 @@ mod verify {
         }
     }
 
-    // pub const unsafe fn new(ptr: *mut T) -> Option<Self>
+    // pub const fn new(ptr: *mut T) -> Option<Self>
     #[kani::proof_for_contract(NonNull::new)]
     pub fn non_null_check_new() {
         let mut x: i32 = kani::any();
         let xptr = &mut x;
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+
+    // pub const unsafe fn add(self, count: usize) -> Self
+    #[kani::proof_for_contract(NonNull::add)]
+    pub fn non_null_check_add() {
+        const SIZE: usize = 100000;
+        // Randomize pointer offset within array bound
+        let offset = kani::any_where(|x| *x <= SIZE as isize && *x >=0);
```

Comment:
Can you simplify this by using a non-deterministic usize and using wrapping_add instead?

Reply:
@celinval This would be the most ideal case however this approach is essentially unbounded and will take too long for kani to execute.

Reply:
Used `PointerGenerator` in the latest commit.